libbeat/processors/add_process_metadata/add_process_metadata.go

```diff
@@ -190,15 +190,29 @@ func (p *addProcessMetadata) enrich(event common.MapStr, pidField string) (resul
 		return nil, errors.Errorf("cannot parse field '%s' (not an integer or string)", pidField)
 	}
 
+	var meta common.MapStr
+
 	metaPtr, err := p.provider.GetProcessMetadata(pid)
 	if err != nil || metaPtr == nil {
+		// no process metadata, lets still try to get container id
 		p.log.Debugf("failed to get process metadata for PID=%d: %v", pid, err)
-		return nil, ErrNoProcess
+		meta = common.MapStr{}
+	} else {
+		meta = metaPtr.fields
 	}
-	meta := metaPtr.fields
 
-	if err = p.enrichContainerID(pid, meta); err != nil {
-		return nil, err
```

Comment:
the first part looks great, but why is this change needed? I think that `p.enrichContainerID` would do what you wanted?

Reply:
a. p.enrichContainerID was function having only side effect - changing external object. Not a good functional style, hard to follow. IMHO.
b. It also set empty cid to meta, why add empty container id to event?

Actually in general I find handling container id separately from other process metadata a bit hard to follow style. Logically it is same as extracting other process metadata, and it is extracted also from same place - files under /proc/pid. So it would fit nicely into go-sysinfo package (iirc) where all other metadata is extracted from process files.

Reply:
@exekias  should i add changelog entry to CHANGELOG.next? 
is it ok to fix also line for main feature we are modifying? this sentence does not seem to be correct in my opinion:
- Add capability of enrich `container.id` with process id in `add_process_metadata` processor {pull}15947[15947]

IMO Container metadata is enriched with container id based on process id.
:)

Reply:
Ok the change LGTM then. About changelog:

This feature was already released so updating the changelog will have no effect. In the same sense, as this is changing the current behaviour we will need a new line :)

Reply:
stuff in changelog.next is released? what file i need to add my line then?

Reply:
@exekias updated changelog as stated in dev guide.